File: app/src/test/java/com/monk/aidldemo/Student.java
package com.monk.aidldemo;

/**
 * @author monk
 * @date 2019-03-22
 */
public class Student extends Person{
    public Student(String name, int age) {
        super(name, age);
    }

    @Override
    public void testParentReturn(int index) {
        super.testParentReturn(index);

        System.out.println("i am student");
    }
}

File: a_kotlin/src/main/java/com/monk/kotlinstudy/extends/Student.kt
package com.monk.kotlinstudy.extends

/*** 主构造函数*/
class Student(name: String, age: Int) : Person() {

    val height: Int = 175

    init {
        println("student init")
    }

    /*** 次构造函数*/
    constructor(name: String) : this("", 12) {}
    constructor() : this("", 12) {}


    companion object{
        @JvmStatic
        fun main(args: Array<String>) {
            val s = Student()

        }
    }

}